the-collab-lab/tcl-15-smart-shopping-list
# Issue #1 - Read from and write to the Firestore database

## Type of Changes

|      | Type                       |
| --- | -------------------------- |
|  ✓    | :sparkles: New feature     |
|  ✓    | :link: Update dependencies |

## Acceptance Criteria


## Related Issue
Closes #1 

## Description 

This PR adds three new components: 
- `ItemsList`: A component responsible for retrieving and displaying the database collection documents on the screen.
- `AddItemsForm`: A component responsible for making changes to the database.
- `Home`: A component works as a main page and a container for both `ItemsList` and `AddItemsForm` components.

### Display items from the database 

Inside that `ItemsList` component, we've used `FirestoreCollection` which is a component that allows you to interact with a Firestore collection. Using this component, you can access the collection at a given path and provide sort options, perform queries, and paginate data.

This component will set up a listener and update whenever our collection is updated in Firestore.

More documentation on `FirestoreCollection` [here](https://www.npmjs.com/package/react-firestore#firestorecollection). 

### Write items to the database

To add items to the database we declared a simple function: `adddToDatabase()` function. Inside that function, we interacted with the `items` collection in the database through `firebase.firestore()` and then passed all the properties we needed to store as a new document in the `items` collection.

### How it looks so far (super ugly)
![a form and a list of a few items retrieved from the database](https://i.ibb.co/Qp7qp8y/style.jpg)

## Testing Steps / QA Criteria

* From your terminal, pull down this branch with `git pull origin bo-hn-connect-firebase` and check that branch out with `git checkout bo-hn-connect-firebase` 
* Then `yarn` to install the newly added dependencies locally and `yarn start` to launch the app.
* Fill out the form with a new item. Make sure it adds it to the list and make sure that the item appears in the `items` collection of the database [here](https://console.firebase.google.com/u/0/project/tcl-15-smart-shopping-list/firestore/data~2F).
